### ISSUE TITLE
RUF013: False positives with type aliases 

### ISSUE THREAD

**Issue by araffin** [astral-sh/ruff#5295]
```python
from typing import Optional

MaybeInt = Optional[int]


def test(a: MaybeInt = None) -> None:
    print(a)
```

`ruff test.py --select RUF`

```
test.py:6:13: RUF013 [*] PEP 484 prohibits implicit `Optional`
Found 1 error.
[*] 1 potentially fixable with the --fix option.
```

ruff version: 0.0.274

found originally in https://github.com/DLR-RM/stable-baselines3.

Is that intended or is that a false positive?

EDIT: I think the problem comes more from the fact that RUF013 is a rule for a type checker, it will be hard to enforce for a linter

**Comment by dhruvmanila** [comment]
Thanks for opening this. This is a known issue.

\cc @charliermarsh Any thoughts on how to go about solving this? 

A simple solution would be to keep track of these aliases and resolve them while checking. This might not work for forward references if the alias is defined after the function definition but I think that should be rare.

Another option would be to check if the top level type is a builtin type or coming from the `typing` module, if not then we can ignore flagging such instances. This would lead to false negatives but they're better than false positives.

**Comment by charliermarsh** [comment]
@dhruvmanila - Maybe we refine RUF013 so that it ignores any types we can't resolve? Like it only operates on primitives and things in the typing module.

**Comment by dhruvmanila** [comment]
So, for nested types we would just consider it as `Any`, thus not flagging it.

A false negative example which I think we should document it in the rule:

```python
Text = str | byte


def f(arg: Text = None):
	pass
```

**Comment by charliermarsh** [comment]
Yeah, we should continue to flag:

```python
def f(arg: Union[str, byte] = None): ...
```

But not:

```python
Text = str | byte


def f(arg: Text = None):
	pass
```

**Comment by charliermarsh** [comment]
(Aliases within the same file, I think we can fix, if we want... Aliases across files we can't _yet_ fix.)

**Comment by charliermarsh** [comment]
@dhruvmanila - Do you want to work on this now and include it in today's release? Or wait till the next release? Up to you, just want to decide if I should wait for it :)

**Comment by dhruvmanila** [comment]
Please go ahead with today's release. I'll work on it tomorrow.

**Comment by Avasam** [comment]
Since there's no meta-issue about multi-file analysis, should this be re-open with the `label:multifile-analysis` ?

Today I found 3 false-positives in typeshed using `logging._ExcInfoType` and `sys._ExitCode`
